### PR TITLE
Add Scan Report Tables view

### LINKED ISSUE #705: Scan Report Tables
Rewrite this view with Nextjs.

Example: /scanreports/226/tables/840

Frontend Tasks:

 - [ ] Recreate the existing UI
 - [ ] Define types in Typescript for what is needed
 - [ ] Add pagination
 - [ ] Add filtering
 - [ ] Add sorting
 - [ ] Add the "Add Concept" POST
 - [ ] Review

Backend Tasks:

 - [ ] New API endpoint
 - [ ] With pagination, filtering, and sorting
 - [ ] Testing
 - [ ] Add the URLS to proxy it
 Review

### PR DESCRIPTION
**Pull request by AndrewThien**
This update adds the UI of the Scan Reports Tables page and changes the following Backend:

- Making V2 of the ScanReportTables API, thanks to @AndyRae 
- Defined the columns and data types to render the list of Tables of selected Scan report
- Added API endpoints to get the Tables list and the "Parent" Scan report name
- Added Proxy URLs (in app/api/proxy/urls.py)
- Implemented the server-side pagination, sorting and filtering
- Add links to the buttons
- Add "linkPrefix" Prop for the data table

Close #705 

# Screenshots

- UI of the page
Screenshot 2024-05-08 at 13 00 20

# Checks

**Important:** please complete these **before** merging.
- [x] Run migrations, if any.
- [x] Update `changelog.md`, including migration instructions if any.
- [x] Run unit tests.

### CODE REVIEW

**Review by dcl10 — CHANGES_REQUESTED** [review]

File: app/next-client-app/app/datasets/page.tsx [hunk 1]
```diff
@@ -67,6 +67,7 @@ export default async function DataSets({ searchParams }: DataSetListProps) {
               data={dataset.results}
               count={dataset.count}
               filter="name"
+              linkPrefix=""
```
Comment (Dev-Akashili) [inline app/next-client-app/app/datasets/page.tsx:70]: You can take this out as well as the other empty string `linkPrefix`

File: app/next-client-app/components/data-table/index.tsx [hunk 2]
```diff
@@ -37,13 +37,19 @@ interface DataTableProps<TData, TValue> {
   data: TData[];
   count: number;
   filter: string;
+  linkPrefix: string;
```
Comment (Dev-Akashili) [inline app/next-client-app/components/data-table/index.tsx:40]: Can make this `linkPrefix?: string;`

File: app/next-client-app/app/datasets/page.tsx (hunk 1 above)
Comment (AndrewThien) [reply app/next-client-app/app/datasets/page.tsx:70]: If I remove this, the URL on the browser will show "undefined" for this part. So I needed it there, I guess. Can you please check on your end as well?

File: app/next-client-app/components/data-table/index.tsx [hunk 3]
```diff
@@ -37,13 +37,19 @@ interface DataTableProps<TData, TValue> {
...
+function UrlBuider(id: string, prefix: string = "") {
+  return `${prefix}${id}/`;
 }
 
 export function DataTable<TData, TValue>({
   columns,
   data,
   count,
   filter,
+  linkPrefix,
```
Comment (Dev-Akashili) [inline app/next-client-app/components/data-table/index.tsx:52]: `linkPrefix=""`

**Review by dcl10 — APPROVED** [review]

**Review by Dev-Akashili — APPROVED** [review]

**Review by dcl10 — APPROVED** [review]

File: app/api/api/views.py [hunk 4]
```diff
@@ -606,6 +610,28 @@ def partial_update(self, request: Any, *args: Any, **kwargs: Any) -> Response:
...
+    filterset_fields = {
+        "scan_report": ["in", "exact"],
+        "name": ["in", "icontains"],
+        "id": ["in", "exact"],
+    }
+    filter_backends = [DjangoFilterBackend, OrderingFilter]
+    ordering_fields = ["name", "person_id", "date_event"]
+    pagination_class = CustomPagination
+
+    ordering = "-created_at"
```
Comment (AndyRae) [inline app/api/api/views.py:623]: this doesn't exist on the model, so won't be used.

File: app/next-client-app/api/scanreports.ts [hunk 5]
```diff
@@ -5,24 +5,51 @@ import request from "@/lib/api/request";
...
-      ),
-    );
+    return await request<ScanReport>(fetchKeys.list(filter));
   } catch (error) {
     console.warn("Failed to fetch data.");
     return { count: 0, next: null, previous: null, results: [] };
   }
 }
 
+export async function getScanReportName(id: string): Promise<ScanReportResult> {
```
Comment (AndyRae) [inline app/next-client-app/api/scanreports.ts:36]: Should just be called `getScanReport`

File: app/next-client-app/api/scanreports.ts [hunk 6]
```diff
@@ -5,24 +5,51 @@ import request from "@/lib/api/request";
 const fetchKeys = {
   list: (filter?: string) =>
     filter ? `v2/scanreports/?${filter}` : "v2/scanreports",
+  tables: (filter?: string) => `v2/scanreporttables/?${filter}`,
+  scanReport: (id: string) => `v2/scanreports/${id}/`,
   update: (id: number) => `scanreports/${id}/`,
 };
 
+export async function getScanReportsTables(
```
Comment (AndyRae) [inline app/next-client-app/api/scanreports.ts:13]: `getScanReportTables`

File: app/next-client-app/types/scanreport.ts [hunk 7]
```diff
@@ -13,3 +13,20 @@ interface ScanReport {
   previous: string | null;
   results: ScanReportResult[];
 }
+
+interface ScanReportTables {
```
Comment (AndyRae) [inline app/next-client-app/types/scanreport.ts:17]: At this point - I feel like these pagination types should be just combined. And they take the result type. The result type should then be the name of the model, not +`Result`

Comment (AndrewThien) [reply app/next-client-app/types/scanreport.ts:17]: Thanks, Andy.
The changes based on your comments will definitely be made on the branch Scanreport-fields.

**Pull request merged**